exasol_error_reporting_python/error_message_builder.py

```diff
@@ -0,0 +1,129 @@
+import re
+from typing import Any
+from exasol_error_reporting_python.parameters_mapper import ParametersMapper
+from exasol_error_reporting_python.placeholder_handler import PlaceholderHandler
+
+ERROR_CODE_FORMAT = "^[FEW]-[[A-Z][A-Z0-9]+(-[A-Z][A-Z0-9]+)*-[0-9]+$"
+
+
+class ErrorMessageBuilder:
+    """
+    This class is a builder for Exasol error messages.
+    """
+
+    def __init__(self, error_code: str):
+        assert re.compile(ERROR_CODE_FORMAT).match(error_code)
+
+        self._error_code = error_code
+        self._message_builder = []
+        self._mitigations = []
+        self._parameter_dict = {}
+
+    def message(self, message: str, *arguments):
+        """
+        Add the given error message to the error builder list and invokes
+        the method that perform mapping of the given parameters with the
+        placeholders in the message.
```

Comment:
maybe mention, then multiple calls will append the given message to the previous ones